### ISSUE TITLE
Stat & BarGauge & Gauge: Color mode option none 

### ISSUE THREAD

**Issue by hannahchan** [grafana/grafana#23223]
**What would you like to be added**:

I'm currently trying out the new Stat panel in Grafana v6.7.1 and have created the following dashboard.

![Screen Shot 2020-03-31 at 9 24 31 pm](https://user-images.githubusercontent.com/22674140/78017400-f5d73880-7397-11ea-92b6-2b595924b60d.png)

However when I switch to the light theme, the numbers in the new Stat panel disappear.

![Screen Shot 2020-03-31 at 9 24 41 pm](https://user-images.githubusercontent.com/22674140/78017514-1d2e0580-7398-11ea-80af-34ead46ffa3b.png)

This is because the base color is set to white. The inverse is also true when you set the base color to black and switch to the dark theme.

Can we add an **Auto** color that can automatically switch between black and white depending on the theme that is selected?

**Why is this needed**:
This is only required to make the panel more usable in light and dark themes when the base color is black or white.

**Comment by dprokop** [comment]
@hannahchan thanks for reporting this. Unfortunately I cannot reproduce this any longer. Can you try upgrading Grafana to 6.7.2?

**Comment by hannahchan** [comment]
@dprokop I just tried with Grafana 6.7.2 and it still appears to be the case. Basically when I set the setting below to white, it looks bad in the light theme.

![Screen Shot 2020-04-21 at 8 54 59 pm](https://user-images.githubusercontent.com/22674140/79858284-868eba80-8412-11ea-8454-1f768e6052e4.png)

My current workaround to make the dashboard look alright in both the dark and the light theme is to set the base colour to grey.

Below is the JSON of a test dashboard.

```json
{
  "annotations": {
    "list": [
      {
        "builtIn": 1,
        "datasource": "-- Grafana --",
        "enable": true,
        "hide": true,
        "iconColor": "rgba(0, 211, 255, 1)",
        "name": "Annotations & Alerts",
        "type": "dashboard"
      }
    ]
  },
  "editable": true,
  "gnetId": null,
  "graphTooltip": 0,
  "id": 2,
  "links": [],
  "panels": [
    {
      "datasource": "Prometheus",
      "gridPos": {
        "h": 7,
        "w": 3,
        "x": 0,
        "y": 0
      },
      "id": 2,
      "options": {
        "colorMode": "value",
        "fieldOptions": {
          "calcs": [
            "lastNotNull"
          ],
          "defaults": {
            "mappings": [],
            "thresholds": {
              "mode": "absolute",
              "steps": [
                {
                  "color": "white",
                  "value": null
                }
              ]
            },
            "unit": "s"
          },
          "overrides": [],
          "values": false
        },
        "graphMode": "none",
        "justifyMode": "auto",
        "orientation": "auto"
      },
      "pluginVersion": "6.7.2",
      "targets": [
        {
          "expr": "sum(rate(http_request_duration_seconds_sum{job=~\"${job:regex}\", controller=~\"${controller:regex}\", action=~\"${action:regex}\", method=~\"${method:regex}\", code=~\".+\"}[$__range])) /\nsum(rate(http_request_duration_seconds_count{job=~\"${job:regex}\", controller=~\"${controller:regex}\", action=~\"${action:regex}\", method=~\"${method:regex}\", code=~\".+\"}[$__range]))",
          "instant": true,
          "legendFormat": "Average Request Duration",
          "refId": "A"
        },
        {
          "expr": "sum(irate(http_request_duration_seconds_sum{job=~\"${job:regex}\", controller=~\"${controller:regex}\", action=~\"${action:regex}\", method=~\"${method:regex}\", code=~\".+\"}[$__range])) /\nsum(irate(http_request_duration_seconds_count{job=~\"${job:regex}\", controller=~\"${controller:regex}\", action=~\"${action:regex}\", method=~\"${method:regex}\", code=~\".+\"}[$__range]))",
          "instant": true,
          "legendFormat": "Instant Request Duration",
          "refId": "B"
        }
      ],
      "type": "stat"
    }
  ],
  "schemaVersion": 22,
  "style": "dark",
  "tags": [],
  "templating": {
    "list": []
  },
  "time": {
    "from": "now-6h",
    "to": "now"
  },
  "timepicker": {
    "refresh_intervals": [
      "5s",
      "10s",
      "30s",
      "1m",
      "5m",
      "15m",
      "30m",
      "1h",
      "2h",
      "1d"
    ]
  },
  "timezone": "",
  "title": "New dashboard Copy",
  "uid": "az-73A3Wk",
  "variables": {
    "list": []
  },
  "version": 2
}
```

**Comment by aloababa** [comment]
Well this is still present in Grafana 7.1. Reopen this issue maybe ?

**Comment by torkelo** [comment]
White is not a good threshold color if you want to support both themes

**Comment by hannahchan** [comment]
I updated Grafana to 7.1.1 and the Stats Plugin version to 7.1.1 as well and can confirm that it's still happening in that version. I quite like the new Stats panel as I find it more visually pleasing however in the dashboard above, I don't need the threshold colors to change.

To support the light theme, I've been using the `grey` color as it's the only neutral color that works in both themes. If I use any other color other than white, black or grey, the dashboard starts to get confusing as colors on a dashboard usually mean something. For example, if you set the color to green, people may automatically think the Stat is good. If set to red then it's bad. Yellow doesn't work either and blue is kind of neutral depending on context.

It's even worst when there is a mixture of colors in a single dashboard. For example, if there are two Stat panels on a dashboard and one of them presents a neutral metric and the other presents a metric using the threshold colors, red, yellow and green. What color do you set the neutral Stat panel to when you want to support dark and light themes? 

I use Grafana primarily as a communication tool to convey system health. I believe that communication is much more effective when viewers have less to noise to reason with.

![Screen Shot 2020-08-01 at 9 53 34 pm](https://user-images.githubusercontent.com/22674140/89101334-5360bc00-d442-11ea-928c-706181848b58.png)

![Screen Shot 2020-08-01 at 9 54 03 pm](https://user-images.githubusercontent.com/22674140/89101372-9a4eb180-d442-11ea-9464-0646e01241a6.png)

![Screen Shot 2020-08-01 at 9 54 27 pm](https://user-images.githubusercontent.com/22674140/89101361-8905a500-d442-11ea-98df-9b0807f330ba.png)

![Screen Shot 2020-08-01 at 9 54 50 pm](https://user-images.githubusercontent.com/22674140/89101363-8acf6880-d442-11ea-90e8-eb0eef447fef.png)

**Comment by torkelo** [comment]
Think a color mode None ,where no coloring is used and value is just using default text color would solve this

**Comment by AdmiralNemo** [comment]
Could this issue be reopened?  It definitely exists in 7.3.4.  It would be nice to have an option for the same behavior as the original Single Stat panel, where the color changes based on the theme, not on the value.

**Comment by Yentel** [comment]
Still present, as said above, an option to automatically change the color of a panel according to the theme would be ideal.
Right now we are forced to use "a color".

Also, (not related to this specific issue) if you use a stat panel with a table query (to display the contents of a label from a prometheus metric) the treshold color is ignored and green seems to be enforced.

**Comment by ashharrison90** [comment]
this is definitely fixed in 8.1 (and was probably available in 8.0) via the `Text color` option:

![image](https://user-images.githubusercontent.com/20999846/128850945-2ffd053a-348a-47b1-8755-d83c960f2fa8.png)
